### PR TITLE
add extraction condition methods in SqlEntityQuery

### PR DESCRIPTION
**Pull request by HidekiSugimoto189**
refs #152
limit and offset method is not implemented.

@shout-star @ota-meshi 
Please review.

### CODE REVIEW

File: src/main/java/jp/co/future/uroborosql/SqlEntityQueryImpl.java [hunk 1]
```diff
@@ -75,11 +97,1002 @@
...
+	}
+
+	/**
+	 * {@inheritDoc}
+	 *
+	 * @see jp.co.future.uroborosql.fluent.SqlEntityQuery#where(java.lang.CharSequence)
+	 */
+	@Override
+	public SqlEntityQuery<E> where(final CharSequence rawString) {
+		this.rawString = rawString;
```
Comment (shout-star) [inline src/main/java/jp/co/future/uroborosql/SqlEntityQueryImpl.java]: I'd like you to call `where`, multiple times.

File: src/main/java/jp/co/future/uroborosql/SqlEntityQueryImpl.java [hunk 2]
```diff
@@ -75,11 +97,1002 @@
...
+	}
+
+	/**
+	 * {@inheritDoc}
+	 *
+	 * @see jp.co.future.uroborosql.fluent.SqlEntityQuery#orderByAsc(java.lang.String[])
+	 */
+	@Override
+	public SqlEntityQuery<E> orderByAsc(final String... cols) {
+		this.orderByColumns = Arrays.asList(cols);
```
Comment (shout-star) [inline src/main/java/jp/co/future/uroborosql/SqlEntityQueryImpl.java]: I'd like you to call `orderByAsc` and `orderByDesc`, multiple times, too.
Can not correspond to `nulls first `and `nulls last`?

File: src/main/java/jp/co/future/uroborosql/SqlEntityQueryImpl.java (hunk 1 above)
Comment (HidekiSugimoto189) [reply src/main/java/jp/co/future/uroborosql/SqlEntityQueryImpl.java]: I agree.
I think that it is better to be able to specify more than once so I will fix it

File: src/main/java/jp/co/future/uroborosql/SqlEntityQueryImpl.java (hunk 2 above)
Comment (HidekiSugimoto189) [reply src/main/java/jp/co/future/uroborosql/SqlEntityQueryImpl.java]: Change the API as follows. And asc () / desc () to be called multiple times.

```
/**
 * multi columns. nulls last(default)
 */
asc(String... cols)
```
```
/**
 * single column. pass nulls option(Nulls.FIRST / Nulls.LAST). 
 */
asc(String col, Nulls nulls)
```

File: src/main/java/jp/co/future/uroborosql/SqlEntityQueryImpl.java [hunk 3]
```diff
@@ -75,11 +91,598 @@
...
+	public SqlEntityQuery<E> in(final String col, final Object... values) {
+		context().param(col, new In(col, values));
+		return this;
+	}
+
+	/**
+	 * @see jp.co.future.uroborosql.fluent.SqlEntityQuery#in(java.lang.String, java.util.List)
+	 */
+	@Override
+	public SqlEntityQuery<E> in(final String col, final List<Object> valueList) {
```
[the remark this answers is not in the document]
Comment (HidekiSugimoto189) [reply src/main/java/jp/co/future/uroborosql/SqlEntityQueryImpl.java]: Thank you for review.
Fixed.

File: src/main/java/jp/co/future/uroborosql/SqlEntityQueryImpl.java [hunk 4]
```diff
@@ -75,11 +91,598 @@
...
+	public SqlEntityQuery<E> notIn(final String col, final Object... values) {
+		context().param(col, new NotIn(col, values));
+		return this;
+	}
+
+	/**
+	 * @see jp.co.future.uroborosql.fluent.SqlEntityQuery#notIn(java.lang.String, java.util.List)
+	 */
+	@Override
+	public SqlEntityQuery<E> notIn(final String col, final List<Object> valueList) {
```
[the remark this answers is not in the document]
Comment (HidekiSugimoto189) [reply src/main/java/jp/co/future/uroborosql/SqlEntityQueryImpl.java]: Thank you for review.
Fixed.

File: src/main/java/jp/co/future/uroborosql/SqlEntityQueryImpl.java [hunk 5]
```diff
@@ -75,11 +91,598 @@
...
+	public SqlEntityQuery<E> like(final String col, final String searchValue, final boolean suffix) {
+		context().param(col, new Like(col, searchValue, suffix));
+		return this;
+	}
+
+	/**
+	 * @see jp.co.future.uroborosql.fluent.SqlEntityQuery#like(java.lang.String, boolean, java.lang.String, boolean)
+	 */
+	@Override
+	public SqlEntityQuery<E> like(final String col, final boolean prefix, final String searchValue,
```
[the remark this answers is not in the document]
Comment (HidekiSugimoto189) [reply src/main/java/jp/co/future/uroborosql/SqlEntityQueryImpl.java]: Thank you for review.
Fixed.

**Comment by HidekiSugimoto189** [comment]
@shout-star @ota-meshi 

I fixed all the items pointed out in the review.

**Comment by shout-star** [comment]
寿司ゆき:thanks_temaki

**Pull request merged**